<a href="https://colab.research.google.com/github/Abbass-Kane/Site_recette_delicieux/blob/main/Bienvenue_dans_Colaboratory.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [16]:
import cv2 as cv2

In [17]:
matrice = cv.imread("Lenna.png") # charge le fichier dans une matrice de pixels couleur 

In [18]:
print(matrice.shape) # affiche les dimension de la matrice 

(512, 512, 3)


In [19]:
print(matrice[0,0]) # accède à la valeur du premier pixel

[125 137 226]


En effet, l’image chargée par la fonction  imread  fait 512 lignes sur 512 colonnes, avec chaque pixel stocké sur 3 octets (car c’est une image couleur, en triplets RVB). Ensuite s’affiche la valeur du pixel de la ligne 0 colonne 0 qui est le triplet [125 137 226].

OpenCV gère les valeurs des pixels non pas dans l’ordre RVB, mais à l’inverse dans l’ordre BVR, donc 226 est ici la composante de rouge du pixel de coordonnées (0,0).

In [21]:
matG =cv.cvtColor(matrice, cv2.COLOR_BGR2GRAY) # conversion des triplets BVR en gris 

In [22]:
print(matG.shape)
print(matG[0,0])

(512, 512)
162


Cette fois, il s’affiche :

(512, 512)

162

La nouvelle matrice fait aussi 512 lignes par 512 colonnes, mais sur de simples octets. Et la couleur RVB du pixel a été convertie en 162, qui correspond à la luminance (cf. formule de la composante Y dans l’espace des couleurs YUV vu précédemment).

Il existe un paramètre de la fonction  imread  qui permet de charger une image directement dans une matrice de niveaux de gris : la valeur 0 ajoutable en second paramètre équivalent à  cv.IMREAD_GRAYSCALE. Cependant cette fonction n’utilise pas exactement la même formule de conversion en luminance.

In [23]:
matG=cv.imread("Lena.png",0) # charge le fichier dans une matrice de pixel gris